services/federator/src/Federator/ExternalServer.hs

```diff
@@ -47,11 +49,15 @@ import Wire.API.Federation.GRPC.Types
 callLocal :: (Members '[Brig, Embed IO] r) => Request -> Sem r InwardResponse
 callLocal Request {..} = do
   -- FUTUREWORK(federation): before making a request, check the sender domain and only make the call if the allowlist (use Util.federateWith) allows it.
-  (resStatus, resBody) <- brigCall (unwrapMethod method) path query body
-  -- FUTUREWORK(federation): Decide what to do with 5xx statuses
-  let statusW32 = fromIntegral $ HTTP.statusCode resStatus
-      bodyBS = maybe mempty LBS.toStrict resBody
-  pure $ InwardResponseHTTPResponse $ HTTPResponse statusW32 bodyBS
+  (resStatus, resBody) <- brigCall path body
+  pure $ case HTTP.statusCode resStatus of
+    200 -> InwardResponseBody $ maybe mempty LBS.toStrict resBody
+    -- TODO: There is a unit test for this, but Akshay has seen the integration
+    -- test never sees InwardResponseErr, when the error is supposed to be
+    -- returned, the integration test just sees `InwardResponseBody` with empty
+    -- body. Maybe this is a bug in mu-haskell, maybe something is wrong with
+    -- our integration test, let's verify this.
+    code -> InwardResponseErr $ "Invalid HTTP status from component: " <> Text.pack (show code) <> " " <> Text.decodeUtf8 (HTTP.statusMessage resStatus)
```

Comment:
TODO: Add a log statement here as brig should ideally never return something that is not 200.

Reply:
Actually brig can return 400 when parsing the request body fails.

Reply:
In a case of diverging versions and an endpoint not yet existing you also need to expect a 404. In general making real http requests and always assuming a 200 isn't realistic.